From 43bdccbbd84b4d7a4e2f75ff0a6758fc4eb5364d Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:36:35 -0400
Subject: [misc] mei: convert bus code to use dev_groups

Message-id: <1399041470-21330-3-git-send-email-prarit@redhat.com>
Patchwork-id: 79290
O-Subject: [RHEL7.1 PATCH BZ 1088953 02/77] MEI: convert bus code to use dev_groups
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit 32f389ec5689751bae758c9b3e3982b2acb696c1
Author: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date:   Fri Aug 23 14:24:39 2013 -0700

    MEI: convert bus code to use dev_groups

    The dev_attrs field of struct bus_type is going away soon, dev_groups
    should be used instead.  This converts the MEI bus code to use
    the correct field.

    Acked-by: Tomas Winkler <tomas.winkler@intel.com>
    Cc: Samuel Ortiz <sameo@linux.intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/bus.c b/drivers/misc/mei/bus.c
index f5dabb3..cd2033c 100644
--- a/drivers/misc/mei/bus.c
+++ b/drivers/misc/mei/bus.c
@@ -108,11 +108,13 @@ static ssize_t modalias_show(struct device *dev, struct device_attribute *a,
 
  return (len >= PAGE_SIZE) ? (PAGE_SIZE - 1) : len;
 }
+static DEVICE_ATTR_RO(modalias);
 
-static struct device_attribute mei_cl_dev_attrs[] = {
- __ATTR_RO(modalias),
- __ATTR_NULL,
+static struct attribute *mei_cl_dev_attrs[] = {
+ &dev_attr_modalias.attr,
+ NULL,
 };
+ATTRIBUTE_GROUPS(mei_cl_dev);
 
 static int mei_cl_uevent(struct device *dev, struct kobj_uevent_env *env)
 {
@@ -124,7 +126,7 @@ static int mei_cl_uevent(struct device *dev, struct kobj_uevent_env *env)
 
 static struct bus_type mei_cl_bus_type = {
  .name  = "mei",
- .dev_attrs = mei_cl_dev_attrs,
+ .dev_groups = mei_cl_dev_groups,
  .match  = mei_cl_device_match,
  .probe  = mei_cl_device_probe,
  .remove  = mei_cl_device_remove,
-- 
1.7.1